src/modules/rlm_pap/rlm_pap.c

```diff
@@ -375,11 +401,19 @@ static rlm_rcode_t CC_HINT(nonnull) mod_authorize(void *instance, REQUEST *reque
 
 #ifdef HAVE_OPENSSL_EVP_H
 		case PW_SHA2_PASSWORD:
+		case PW_SSHA224_PASSWORD:
+		case PW_SSHA256_PASSWORD:
+		case PW_SSHA384_PASSWORD:
+		case PW_SSHA512_PASSWORD:
```

Comment:
Unfortunately the arbitrary length of the appended salt makes precise calculation of the algorithm by the resultant overall size indeterminate. I initially tried to blind detect the hash, but this seemed crude vs. the exact attribute definition.